x/auth/tx/config.go

```diff
@@ -76,26 +80,27 @@ var DefaultSignModes = []signingtypes.SignMode{
 // We prefer to use depinject to provide client.TxConfig, but we permit this constructor usage. Within the SDK,
 // this constructor is primarily used in tests, but also sees usage in app chains like:
 // https://github.com/evmos/evmos/blob/719363fbb92ff3ea9649694bd088e4c6fe9c195f/encoding/config.go#L37
-func NewTxConfig(protoCodec codec.Codec, enabledSignModes []signingtypes.SignMode,
-	customSignModes ...txsigning.SignModeHandler,
+func NewTxConfig(protoCodec codec.Codec, enabledSignModes []signingtypes.SignMode, addressPrefix,
+	validatorPrefix string, customSignModes ...txsigning.SignModeHandler,
```

Comment:
Not a fan of this api change as you only need this for the default.
I don't really have a better solution, but I find the api ugly

Reply:
yes, I think the same way. Still looking for an alternative :S

Reply:
I see. What do you think of this function signature:
`func NewTxConfig(protoCodec codec.Codec, addressCodec, validatorAddressCodec address.codec,  enabledSignModes []signingtypes.SignMode, customSignModes ...txsigning.SignModeHandler,
) client.TxConfig`

Then in `NewTxConfigWithOptions` we can return an error if `configOptions.SigningOptions == nil`

This way we make sure that codecs are always provided.

Reply:
That makes sense total sense!

Note, then here, we can use the codec from the signing context on the interface registry: https://github.com/cosmos/cosmos-sdk/blob/a248d05f70f4ad7b8ff7b521e3d23086867d07dc/testutil/sims/simulation_helpers.go#L58, no need to add extra arguments.

Reply:
> Note, then here, we can use the codec from the signing context on the interface registry:

Not sure if I follow you. Do you mean to use the interface registry of the `runtime.AppSimI`. That'd imply a couple of type assertions.

Reply:
No, use it from the interface registry signing context